### ISSUE TITLE
Update SGD regressor loss values for scikit learn 1.0

### ISSUE THREAD

**Issue by eddiebergman** [automl/auto-sklearn#1334]
We need to update the loss name in SGD regressor when switching to scikit learn 1.0.

* [Docs](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDRegressor.html#sklearn-linear-model-sgdregressor)

```python
# Current
loss = CategoricalHyperparameter(
    name="loss",
    choices=[
        "squared_loss", "huber", "epsilon_insensitive", "squared_epsilon_insensitive"
    ],
    default_value="squared_loss",
)

# New
loss = CategoricalHyperparameter(
    name="loss",
    choices=[
        "squared_error", "huber", "epsilon_insensitive", "squared_epsilon_insensitive"
    ],
    default_value="squared_loss",
)
```

> Deprecated since version 1.0: The loss ‘squared_loss’ was deprecated in v1.0 and will be removed in version 1.2. Use loss='squared_error' which is equivalent.
